### PR TITLE
FeatureToggles: Write enabled flags to the logs on startup

### PR DESCRIPTION
**Pull request by ryantxu**
This writes the enabled feature toggles to the logfile on startup.  For me, this looks like:

```
INFO [05-02|16:08:36] App mode development                     logger=settings
INFO [05-02|16:08:36] FeatureToggles                           logger=featuremgmt kubernetesPlaylists=true logRowsPopoverMenu=true cloudWatchNewLabelParsing=true annotationPermissionUpdate=true prometheusDataplane=true dashgpt=true topnav=true lokiQuerySplitting=true prometheusConfigOverhaulAuth=true lokiQueryHints=true alertingInsights=true alertingSimplifiedRouting=true grafanaAPIServerEnsureKubectlAccess=true angularDeprecationUI=true betterPageScrolling=true lokiStructuredMetadata=true recoveryThreshold=true logsContextDatasourceUi=true cloudWatchCrossAccountQuerying=true lokiMetricDataplane=true nestedFolderPicker=true sqlExpressions=true alertingNoDataErrorExecution=true exploreContentOutline=true idForwarding=true grafanaAPIServerWithExperimentalAPIs=true panelMonitoring=true transformationsRedesign=true returnToPrevious=true prometheusMetricEncyclopedia=true exploreMetrics=true nestedFolders=true awsAsyncQueryCaching=true recordedQueriesMulti=true influxdbBackendMigration=true logsExploreTableVisualisation=true publicDashboards=true dataplaneFrontendFallback=true queryService=true enableElasticsearchBackendQuerying=true managedPluginsInstall=true
INFO [05-02|16:08:36] Connecting to DB                         logger=sqlstore dbtype=sqlite3
INFO [05-02|16:08:36] Locking database                         logger=migrator
```

Or an earlier version could look like:
```
INFO [05-02|14:47:32] App mode development                     logger=settings
INFO [05-02|14:47:32] FeatureToggles                           logger=featuremgmt enabled="[annotationPermissionUpdate prometheusDataplane transformationsRedesign dataplaneFrontendFallback lokiStructuredMetadata prometheusConfigOverhaulAuth recoveryThreshold lokiQueryHints kubernetesPlaylists alertingInsights angularDeprecationUI exploreMetrics lokiMetricDataplane cloudWatchNewLabelParsing panelMonitoring managedPluginsInstall exploreContentOutline logsExploreTableVisualisation publicDashboards sqlExpressions queryService nestedFolders prometheusMetricEncyclopedia lokiQuerySplitting enableElasticsearchBackendQuerying grafanaAPIServerEnsureKubectlAccess alertingNoDataErrorExecution recordedQueriesMulti logRowsPopoverMenu topnav alertingSimplifiedRouting awsAsyncQueryCaching nestedFolderPicker betterPageScrolling logsContextDatasourceUi dashgpt returnToPrevious idForwarding influxdbBackendMigration cloudWatchCrossAccountQuerying grafanaAPIServerWithExperimentalAPIs]"
INFO [05-02|14:47:32] Connecting to DB                         logger=sqlstore dbtype=sqlite3
INFO [05-02|14:47:32] Locking database                         logger=migrator
```

### CODE REVIEW

**Comment by ryantxu** [comment]
We should pick some pattern that makes a loki query as easy as possible.  This will be written out once at startup

**Comment by bergquist** [comment]
I think it might be easier if we write the toggleName=ToogleValue as keys.

That would make it possible to write loki queries for certain feature toggles. 

`{job="hosted-grafana/grafana"} |= "FeatureToggles" | logfmt | grafanaapiserver=true`

**Review by bergquist — APPROVED** [review]
LGTM

File: pkg/services/featuremgmt/service.go [hunk 1]
```diff
@@ -57,6 +60,15 @@ func ProvideManagerService(cfg *setting.Cfg) (*FeatureManager, error) {
 	// update the values
 	mgmt.update()
 
+	// Log the enabled feature toggles at startup
+	enabled := sort.StringSlice(maps.Keys(mgmt.enabled))
+	logctx := make([]any, len(enabled)*2)
+	for i, k := range enabled {
+		logctx[(i * 2)] = k
+		logctx[(i*2)+1] = true
```
Comment (hairyhenderson) [inline pkg/services/featuremgmt/service.go:68]: I think it'd be (optionally? maybe?) valuable to also log toggles which are present but disabled, but that's not necessary right away for this.

**Review by hairyhenderson — APPROVED** [review]

**Pull request merged**